src/MgvOfferMaking.sol

```diff
@@ -304,7 +304,13 @@ contract MgvOfferMaking is MgvHasOffers {
       require(TickLib.inRange(insertionTick), "mgv/writeOffer/tick/outOfRange");
 
       // must cache tick because branch will be modified and tick information will be lost (in case an offer will be removed)
-      Tick cachedLocalTick = ofp.local.bestTick();
+      Tick cachedLocalTick;
+      // force control flow through gas saving path if offer list is empty
+      if (ofp.local.level0().isEmpty()) {
+        cachedLocalTick = insertionTick;
+      } else {
+        cachedLocalTick = ofp.local.bestTick();
+      }
 
       // remove offer from previous position
       if (ofp.oldOffer.isLive()) {
```

Comment:
These if-stmts can be merged: If the offer list is empty, the offer cannot be live.

Reply:
But the implication is not true the other way? How would you write it

Reply:
```suggestion
      if (ofp.local.level0().isEmpty()) {
        cachedLocalTick = insertionTick;
      } else
        cachedLocalTick = ofp.local.bestTick();
        // remove offer from previous position
        if (ofp.oldOffer.isLive()) {
```

Reply:
nice!

Reply:
nice! addressed in 006d6c3c31462e8ba582050c7baa34a69d9033dc